CAIPs/caip-341.md

```diff
@@ -0,0 +1,125 @@
+---
+caip: 341
+title: Extension ID Target Type Specification
+author: [Joao Tavares] (@ffmcgee725)
+discussions-to: https://github.com/ChainAgnostic/CAIPs/issues/341
+status: Draft
+type: Standard
+created: 2024-12-12
+requires: 294
+---
+
+## Simple Summary
+
+CAIP-341 defines the `extensionId` type as a valid target type for establishing connections with browser extension wallets via the [CAIP-294] `wallet_announce` wallet discovery event.
+
+## Abstract
+
+This proposal introduces a new target type `extensionId` for the `target` field of the `walletData` interface dispatched in [CAIP-294]'s `wallet_announce` event. This target type is used to specify the extension ID of a browser extension wallet, allowing callers to establish connections with the wallet using the [`externally_connectable`][externally_connectable API documentation] API.
+
+## Motivation
+
+CAIP-294 proposes a solution to fragmentation across blockchain ecosystems wallet discovery mechanisms (e.g  Ethereum's [EIP-6963], Solana's [Wallet Standard]). By defining a standardized target type for browser extension wallets that use the `externally_connectable` browser API, we aim to extend CAIP-294's unified solution to cross ecosystem wallet discoverability, enhancing interoperability across these different blockchain ecosystems.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC-2119](https://www.rfc-editor.org/rfc/rfc2119).
+
+### Definitions
+
+Wallet Provider: A user agent that manages accounts and facilitates transactions with a blockchain.
+
+Decentralized Application (dapp): A web page that relies upon one or many Web3 platform APIs which are exposed to the web page via the Wallet.
+
+Blockchain Library: A library or piece of software that assists a dapp to interact with a blockchain and interface with the Wallet.
+
+### Target Type
+
+The `target` field in the `walletData` interface is used to specify the connection method for the wallet. This CAIP introduces the `extensionId` type as a valid target type.
```

Comment:
all of the `walletData` should probably be `WalletData` since we're referencing the interface itself and not a specific instance of the interface

Reply:
resolved in [79df192](https://github.com/ChainAgnostic/CAIPs/pull/341/commits/79df1928de49751d3192cc94d053db1e1e231948)